### PR TITLE
CR-1100943 Corrected the Buffer allocation in case of nodma

### PR DESCRIPTION
**Pull request by venkatp-xilinx**
Corrected the Buffer allocation in case of nodma platform

### CODE REVIEW

File: src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx [hunk 1]
```diff
@@ -540,25 +540,33 @@ namespace xclhwemhal2 {
       {
         //CR 966701: alignment to 4k (instead of mDeviceInfo.mDataAlignment)
         mDDRMemoryManager.push_back(new xclemulation::MemoryManager(it.size, it.base_addr, getpagesize(), it.tag));
+        std::string tag = it.tag;
+        int32_t index = it.index;
+
+        std::size_t found = tag.find("HOST");
+        if (found != std::string::npos) {
+          host_sptag_idx = index;
+        }
```
Comment (stsoe) [inline src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx:549]: Why all the copying to local variables?  Maybe simplify as:
```
if (it.tag.find("HOST") != std::string::npos)
  host_sptag_idx = it.index;
```

File: src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx [hunk 2]
```diff
@@ -540,25 +540,33 @@ namespace xclhwemhal2 {
...
-      for (auto it:mDDRMemoryManager)
+
+      for (auto it : mDDRMemoryManager)
       {
-              std::string tag = it->tag();
+        std::string tag = it->tag();
 
-              if(tag.empty() || tag.find("MBG") == std::string::npos)
-        	      continue;
+        if (tag.empty() || tag.find("MBG") == std::string::npos) {
```
Comment (stsoe) [inline src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx:556]: How about just:
```
if (it->tag().find("MBG") == std::string::npos)
 ...;
```
find on empty string returns std::string::npos, no need to check empty first.

File: src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx [hunk 3]
```diff
@@ -540,25 +540,33 @@ namespace xclhwemhal2 {
...
-        	      {
-			      //add child memories
-			      it->mChildMemories.push_back(it2);
-        	      }
-              }
+        for (auto it2 : mDDRMemoryManager)
+        {
+          if (it2->size() != 0 && it2 != it &&
+            it->start() <= it2->start() &&
+            (it->start() + it->size()) >= (it2->start() + it2->size()))
```
Comment (stsoe) [inline src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx:564]: A comment would be great.

File: src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.h [hunk 4]
```diff
@@ -387,6 +387,7 @@ using addr_type = uint64_t;
       std::string simulatorType;
       std::string sim_path;
       std::map<uint64_t, std::pair<void*, uint64_t> > mHostOnlyMemMap;
+      unsigned host_sptag_idx;
```
Comment (stsoe) [inline src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.h:390]: unsigned int

**Review by stsoe — CHANGES_REQUESTED** [review]

File: src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx (hunk 1 above)
Comment (venkatp-xilinx) [reply src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx]: Removed the local variables as suggested

File: src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx (hunk 2 above)
Comment (venkatp-xilinx) [reply src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx]: Corrected the condition as suggestion

File: src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx (hunk 3 above)
Comment (venkatp-xilinx) [reply src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx:566]: Added comment as requested

File: src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.h (hunk 4 above)
Comment (venkatp-xilinx) [reply src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.h]: corrected the declaration as said

**Review by larry9523 — APPROVED** [review]

**Pull request merged**